tests/test_data.py

```diff
@@ -3228,3 +3229,101 @@ async def test_retrieve_optional_field_no_default(init_dataclasses_and_load_sche
 
     report = await data.Report.get_by_id(report.id)
     assert report.returncode is None
+
+
+@pytest.fixture
+def simple_query_builder():
+    return SimpleQueryBuilder(
+        select_clause="SELECT *",
+        from_clause="FROM table",
+        filter_statements=["column = $1"],
+        values=["value"],
+        limit=10,
+        backward_paging=False,
+        prelude="prelude_query AS (SELECT * FROM prelude_table)",
+        prelude_extra=[
+            "extra_prelude_1 AS (SELECT * FROM extra_table1)",
+            "extra_prelude_2 AS (SELECT * FROM extra_table2)",
+        ],
+    )
+
+
+@pytest.fixture
+def prelude_query_builder():
+    return SimpleQueryBuilder(
+        select_clause="SELECT *",
+        from_clause="FROM prelude_table",
+        filter_statements=["prelude_column = $1"],
+        values=["prelude_value"],
+    )
+
+
+def test_prelude_extra_in_simple_query_builder(simple_query_builder):
+    query, _ = simple_query_builder.build()
```

Comment:
Why not assert values?

Reply:
You mean why not use real tables and execute queries on those?